src/box/txn.c

```diff
@@ -1254,6 +1286,23 @@ txn_on_yield(struct trigger *trigger, void *event)
 	if(!txn_has_flag(txn, TXN_CAN_YIELD)) {
 		txn_rollback_to_svp(txn, NULL);
 		txn_set_flags(txn, TXN_IS_ABORTED_BY_YIELD);
+		return 0;
+	} else if (txn_has_flag(txn, TXN_IS_ABORTED_BY_TIMEOUT)) {
+		txn_rollback_to_svp(txn, NULL);
```

Comment:
Why? When you set the flag, you roll back the transaction so it must be already rolled-back.

Reply:
I do this because even after rollback, if we make an insert, the information about it will be stored. So we roll back the transaction with each yield. Same as in case when memtx_use_mvcc_engine = false.

Reply:
I think we shouldn't process any new statements after a rollback on timeout or yield. Please ask @alyapunov.

Reply:
Fixed